btf/handle.go

```diff
@@ -0,0 +1,60 @@
+package btf
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// HandleIterator allows enumerating BTF blobs loaded into the kernel.
+type HandleIterator struct {
+	id  ID
+	err error
+}
+
+func NewHandleIterator() *HandleIterator {
+	return &HandleIterator{0, nil}
+}
+
+// Next retrieves a handle for the next BTF blob.
+//
+// [Handle.Close] is called if *handle is non-nil to avoid leaking fds.
+//
+// Returns true if another BTF blob was found. Call [HandleIterator.Err] after
+// the function returns false.
+func (it *HandleIterator) Next(handle **Handle) bool {
```

Comment:
Maybe this should just be `Next()`, with handle exposed via `HandleIterator.Handle` or similar? See https://pkg.go.dev/github.com/cilium/ebpf/btf#TypesIterator

Reply:
Turns out that this makes it harder to properly clean up the Handle.